src/app/features/profile/interfaces/user-profile-map.ts

```diff
@@ -0,0 +1,3 @@
+import { UserProfile } from "./user-profile";
+
+export type UserProfileMap = Map<string, UserProfile>
```

Comment:
Opting for a Map here rather than a regular JS object or Record type.  The benefit is when TS strict mode is enabled it will appropriately worn that a selected value from the object could be undefined and enforce handling in that case.